From: Sébastien Noel <twolife@debian.org>
Description: fix FTBFS with GCC-16
Forwarded: no

--- a/src/shared/tools.h
+++ b/src/shared/tools.h
@@ -36,10 +36,7 @@
 
 void *operator new(size_t, bool);
 void *operator new[](size_t, bool);
-inline void *operator new(size_t, void *p) { return p; }
-inline void *operator new[](size_t, void *p) { return p; }
-inline void operator delete(void *, void *) {}
-inline void operator delete[](void *, void *) {}
+#include <new>
 
 #ifdef swap
 #undef swap
